azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -892,10 +892,27 @@ private void insertJVMAargs() {
     final String previousJVMArgs = this.props.get(JavaProcessJob.JVM_PARAMS);
     jobJVMArgs += (previousJVMArgs == null) ? "" : " " + previousJVMArgs;
 
+    // Add useful Java options for java jobs which are provided through properties
+    String javaOpts = insertJavaOptions();
```

Comment:
Could we repurpose `JavaProcessJob.JVM_PARAMS` to add this instead of creating a new property ?

Reply:
The reason to keep it separate is we eventually want to remove this fix. That is why keeping the effect of code as minimal as possible.

Reply:
Will discuss this offline.